### PR TITLE
Features/hull node upgrade

### PR DESCRIPTION
**Pull request by mkawalek**

### CODE REVIEW

File: .idea/jsLibraryMappings.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaloo) [inline .idea/jsLibraryMappings.xml]: File added by mistake?

File: README.md [hunk 2]
```diff
@@ -30,3 +30,6 @@ docker-compose run install
 docker-compose up -d redis
 docker-compose up dev # with autoreloading enabled
 ```
+
+### Testing :
+- create developer account at https://developers.hubspot.com/docs/overview
```
Comment (michaloo) [inline README.md]: and maybe add more information to tell to obtain `CLIENT_ID` and `CLIENT_SECRET` and paste it into .env file

File: package.json [hunk 3]
```diff
@@ -27,9 +27,9 @@
     "build:server": "babel server -d build",
     "start": "COMBINED=true node build/web",
     "start:dev": "COMBINED=true babel-watch -L server/web",
-    "test": "npm run test:lint",
-    "test:lint": "eslint server",
-    "test:modules": "npm outdated --depth=0",
+    "test": "npm run test:lint && npm run test:units",
+    "test:lint": "./node_modules/.bin/eslint server",
```
Comment (michaloo) [inline package.json]: didn't work without `./node_modules/.bin/`?
According to the `npm` docs the $PATH should be adjusted so only the name of binary should be enough.

File: server/bootstrap.js [hunk 4]
```diff
@@ -9,29 +9,29 @@ import SyncController from "./controller/sync";
 import NotifyController from "./controller/notify";
 import * as newJobs from "./jobs";
 
-import InstrumentationAgent from "./util/instrumentation-agent";
-import KueAdapter from "./util/queue/adapter/kue";
+const { LOG_LEVEL, SHIP_CACHE_MAX = 100, SHIP_CACHE_TTL = 60, KUE_PREFIX = "hull-hubspot", REDIS_URL = "127.0.0.1", SECRET = "1234", PORT = 8082 } = process.env;
```
Comment (michaloo) [inline server/bootstrap.js]: maybe split that to separate lines?

File: server/controller/batch.js [hunk 5]
```diff
@@ -7,35 +7,37 @@ export default class BatchController {
    * @param  {Object} res
    * @return {Promise}
    */
-  handleBatchExtractAction(req, res) {
-    const segmentId = req.query.segment_id || null;
-    return req.shipApp.queueAgent.create("handleBatchExtractJob", {
```
Comment (michaloo) [inline server/controller/batch.js]: That line should be ported to something like `req.hull.enqueue("handleBatchExtractJob", {})`.

File: server/controller/sync.js [hunk 6]
```diff
@@ -4,46 +4,47 @@
  */
 export default class SyncStrategy {
 
-  syncAction(req, res, next) {
-    return req.shipApp.queueAgent.create("startSyncJob")
+  syncAction = (req, res, next) => {
```
Comment (michaloo) [inline server/controller/sync.js]: Mistaken `=` character?

File: server/worker.js [hunk 7]
```diff
@@ -1,30 +1,9 @@
...
-
-function handleExit() {
-  console.log("Exiting... waiting 30 seconds workers to flush");
-  setTimeout(exitNow, 30000);
-  queueAdapter.exit().then(exitNow);
-}
-
-process.on("SIGINT", handleExit);
-process.on("SIGTERM", handleExit);
+  connector.startWorker();
```
Comment (michaloo) [inline server/worker.js]: Do you remember about shipApp middleware?

File: .idea/jsLibraryMappings.xml (hunk 1 above)
Comment (mkawalek) [inline .idea/jsLibraryMappings.xml]: Yes, unfortunately WebStorm always adds this .idea folder and that is why I am putting this in .gitignore file

File: package.json (hunk 3 above)
Comment (mkawalek) [inline package.json]: You're right, it works :)

File: server/controller/sync.js (hunk 6 above)
Comment (mkawalek) [inline server/controller/sync.js]: Actually, eslit gives me error here:
Expected 'this' to be used by class method 'syncAction'
and that leaves us with two options:
- make this method static 
- make syncAction to return a function
And I chose second choice, but after second thoughts I think we should use static method like in other places.

File: server/worker.js (hunk 7 above)
Comment (mkawalek) [inline server/worker.js]: Yes I do but I do not know how shipApp is connected with worker. Do you mean that we should start worker after setting up shipApp middleware ? But I think that this jobs should not be started before setting this middleware.

Comment (michaloo) [inline server/worker.js]: We already solved that on slack? We can apply the `appMiddleware` easily to the worker middleware stack.

File: server/controller/sync.js (hunk 6 above)
Comment (michaloo) [inline server/controller/sync.js]: ok I get it, let's try the static approach.

**Comment by michaloo** [comment]
Replaced by #44.

**Pull request closed without merging**